### PR TITLE
Disallow wrapping prohibited extrinsics in a batch

### PR DESCRIPTION
**Pull request by ekovalev**
Current token economics prohibits certain pre-funded accounts from staking their tokens that are not fully vested (by default, staking of vested tokens is allowed).
This is implemented through blacklisting these accounts as the origin of the `staking::bond()` dispatchable call.
However, the old implementation didn't take into account the fact such calls could be wrapped in a batch (or even a series of nested batches and/or other `pallet-utility` calls) and thereby go through. This PR fixes that.
Similarly, the other signed extension (`DisableValueTransfers`) has been patched. Just to recap, this other filter will be deactivated starting from Vara stage 3.

TODO: a test for `DisableValueTransfers` wouldn't be out of place, however, since the extension is not parametrised with a `RuntimeCall` type it can only be tested against the real Runtime which would require importing unnecessary testing crates. Whether or not making a more generic version of it has any value is questionable since this filter is set for removal pretty soon anyway.

### CODE REVIEW

**Review by ukint-vs — APPROVED** [review]

File: runtime/vara/src/lib.rs [hunk 1]
```diff
@@ -375,10 +375,36 @@ impl pallet_session_historical::Config for Runtime {
 pub struct BondCallFilter;
 impl Contains<RuntimeCall> for BondCallFilter {
     fn contains(call: &RuntimeCall) -> bool {
-        matches!(
```
Comment (breathx) [inline runtime/vara/src/lib.rs:378]: Exactly shared behavior

File: runtime/vara/src/extensions/mod.rs [hunk 2]
```diff
@@ -49,21 +49,44 @@ impl SignedExtension for DisableValueTransfers {
         _: &DispatchInfoOf<Self::Call>,
         _: usize,
     ) -> TransactionValidity {
-        match call {
-            RuntimeCall::Balances(_) => {
-                Err(TransactionValidityError::Invalid(InvalidTransaction::Call))
-            }
+        let predicate = |call: &Self::Call| match call {
```
Comment (breathx) [inline runtime/vara/src/extensions/mod.rs:52]: It's implementation of Contains trait

**Review by breathx — APPROVED** [review]

File: pallets/staking-rewards/src/mock.rs [hunk 3]
```diff
@@ -190,14 +176,48 @@ impl pallet_sudo::Config for Test {
...
+
+impl pallet_utility::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type RuntimeCall = RuntimeCall;
+    type WeightInfo = ();
+    type PalletsOrigin = OriginCaller;
+}
+
 // Filter that matches `pallet_staking::Pallet<T>::bond()` call
 pub struct BondCallFilter;
```
[the remark this answers is not in the document]
Comment (ekovalev) [reply pallets/staking-rewards/src/mock.rs:188]: It has to have access to the variants of the `RuntimeCall` enum from the current runtime. Hence all such filters are defined in the lib.rs of the runtime.

File: pallets/staking-rewards/src/mock.rs [hunk 4]
```diff
@@ -190,14 +176,48 @@ impl pallet_sudo::Config for Test {
...
+        ) {
+            return true;
+        }
+
+        let mut res = false;
+
+        if let RuntimeCall::Utility(pallet_utility::Call::batch { calls }) = call {
+            for c in calls {
+                if matches!(c, &RuntimeCall::Staking(pallet_staking::Call::bond { .. })) {
+                    res = true;
```
[the remark this answers is not in the document]
Comment (ekovalev) [reply pallets/staking-rewards/src/mock.rs]: fair enough

File: runtime/vara/src/extensions/mod.rs [hunk 5]
```diff
@@ -49,21 +49,44 @@ impl SignedExtension for DisableValueTransfers {
...
+            | RuntimeCall::Gear(pallet_gear::Call::send_reply { value, .. }) => !value.is_zero(),
+            _ => false,
+        };
+        if predicate(call) {
+            return Err(TransactionValidityError::Invalid(InvalidTransaction::Call));
+        }
+
+        let mut has_illegal_call = false;
+
+        if let RuntimeCall::Utility(pallet_utility::Call::batch { calls }) = call {
```
[the remark this answers is not in the document]
Comment (ekovalev) [reply runtime/vara/src/extensions/mod.rs]: Yes, that would also have been a possibility. Fixed that.

**Review by ukint-vs — APPROVED** [review]

**Review by breathx — APPROVED** [review]

**Pull request merged**